multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/CheckForOperationsInProgressStep.java

```diff
@@ -30,89 +28,60 @@
 public class CheckForOperationsInProgressStep extends AsyncFlowableStep {
 
     @Inject
-    private ServiceOperationGetter serviceOperationGetter;
+    protected ServiceOperationGetter serviceOperationGetter;
     @Inject
     private ServiceProgressReporter serviceProgressReporter;
 
     @Override
     protected StepPhase executeAsyncStep(ProcessContext context) {
-        List<CloudServiceInstanceExtended> servicesToProcess = getServicesToProcess(context);
-
-        List<CloudServiceInstanceExtended> existingServices = getExistingServices(context.getControllerClient(), servicesToProcess);
-        if (existingServices.isEmpty()) {
-            return StepPhase.DONE;
-        }
-
-        Map<CloudServiceInstanceExtended, ServiceOperation> servicesInProgressState = getServicesInProgressState(context, existingServices);
-        if (servicesInProgressState.isEmpty()) {
+        List<CloudServiceInstanceExtended> existingServicesInProgress = getExistingServiceInProgress(context);
+        if (existingServicesInProgress.isEmpty()) {
             return StepPhase.DONE;
         }
 
         getStepLogger().info(Messages.WAITING_PREVIOUS_OPERATIONS_TO_FINISH);
 
-        Map<String, ServiceOperation.Type> servicesOperationTypes = getServicesOperationTypes(servicesInProgressState);
+        Map<String, ServiceOperation.Type> servicesOperationTypes = getServicesOperationTypes(existingServicesInProgress);
         getStepLogger().debug(Messages.SERVICES_IN_PROGRESS, JsonUtil.toJson(servicesOperationTypes, true));
         context.setVariable(Variables.TRIGGERED_SERVICE_OPERATIONS, servicesOperationTypes);
 
-        List<CloudServiceInstanceExtended> servicesWithData = getListOfServicesWithData(servicesInProgressState);
-        context.setVariable(Variables.SERVICES_DATA, servicesWithData);
+        context.setVariable(Variables.SERVICES_DATA, existingServicesInProgress);
 
         return StepPhase.POLL;
     }
 
-    protected List<CloudServiceInstanceExtended> getServicesToProcess(ProcessContext context) {
-        return Collections.singletonList(context.getVariable(Variables.SERVICE_TO_PROCESS));
-    }
-
-    private List<CloudServiceInstanceExtended> getExistingServices(CloudControllerClient cloudControllerClient,
-                                                                   List<CloudServiceInstanceExtended> servicesToProcess) {
-        return servicesToProcess.parallelStream()
-                                .map(service -> getExistingService(cloudControllerClient, service))
-                                .filter(Objects::nonNull)
-                                .collect(Collectors.toList());
+    protected List<CloudServiceInstanceExtended> getExistingServiceInProgress(ProcessContext context) {
```

Comment:
The function should be called getExistingServicesInProgress
because it returns multiple services